src/Services/GTT/shared/GTT.Infrastructure/Repositories/ClassRepository.cs

```diff
@@ -49,25 +53,95 @@ public async Task<int> DeleteAsync(int id)
         {
             var sql = @"UPDATE Challenge SET something equal something";
             var result = await _connection.ExecuteAsync(sql, entity, _tran);
             return result;
         }
+
+        public async Task<BaseResponseModel> CreateClass(CreateClassRequestModel request)
+        {
+            try
+            {
+                #region  Check CommunityId and CoachId
+                var sql = @"SELECT cm.CommunityId, c.CoachId  FROM Community cm, Coach c 
+                            WHERE cm.CommunityId = @communityId and c.CoachId = @coachId";
+
+                var parameter = new DynamicParameters();
+
+                parameter.Add("@coachId", request.CoachId);
+                parameter.Add("@communityId", request.CommunityId);
+
+                var queryData = await _connection.QueryFirstOrDefaultAsync<CoachCommunityResponse>(sql, parameter, _tran);
+
+                if (queryData == null)
+                {
+                    return new BaseResponseModel(HttpStatusCode.NotFound, "CommunityId or CoachId invalid");
+                }
+
+                sql = @"SELECT * FROM Class c WHERE c.Title = @title";
+
+                parameter.Add("@title", request.Title);
+                //parameter.Add("@upCase", request.Title.ToLower());
+                //parameter.Add("@upCase", request.Title.ToUpper());
+
+                var queryTitle = await _connection.QueryFirstOrDefaultAsync(sql, parameter, _tran);
+
+                if(queryTitle != null)
+                {
+                    return new BaseResponseModel(HttpStatusCode.BadRequest, "Title must be unique");
+                }
+
+                #endregion
+
+                sql = @"INSERT INTO Class
+                             VALUES(@title, @coachId, @communityId, @duration, @startDate, @createdBy, @updatedBy, @createdDate, @updatedDate, @IsActive);
+                             DECLARE @classId int                             
+                             SET @classId = SCOPE_IDENTITY()
+                             SELECT * FROM Class WHERE ClassId = @classId";
+
+                parameter.Add("@title", request.Title);
+                parameter.Add("@duration", request.Duration);
+                parameter.Add("@startDate", request.StartDate);
+                parameter.Add("@isActive", request.IsActive);
+                parameter.Add("@startDate", request.StartDate);
+                parameter.Add("@createdBy", request.CreatedBy);
+                parameter.Add("@updatedBy", request.UpdatedBy);
+                parameter.Add("@createdDate", request.CreatedDate);
+                parameter.Add("@updatedDate", request.UpdatedDate);
+                parameter.Add("@IsActive", request.IsActive);
+
+                var result = await _connection.QueryFirstAsync<ClassVM>(sql, parameter, _tran);
+
+                _tran.Commit();
+
+                return new BaseResponseModel(HttpStatusCode.OK, "Success", result);
```

Comment:
Remove redundant (HttpStatusCode.OK, "Success"), It should be "return new BaseResponseModel(result); is enough